File: ServiceCore/Translators/UnitFinishingOptionTranslator.cs
using BOCore;
using DALCore.Models;

namespace ServiceCore.Translators
{
    public class UnitFinishingOptionTranslator
    {
        public static ErrorCode TranslateEntityToBO(UnitFinishingOption entity, UnitFinishingOptionBO bo)
        {
            if (entity == null) return ErrorCode.EntityNull;
            if (bo == null) return ErrorCode.BoNull;
            bo.Id = entity.Id;
            bo.UnitId = entity.UnitId;
            bo.Name = entity.Name;
            bo.SortOrder = entity.SortOrder;
            bo.IsDefault = entity.IsDefault;
            return ErrorCode.Success;
        }

        public static ErrorCode TranslateBOToEntity(UnitFinishingOption entity, UnitFinishingOptionBO bo)
        {
            if (entity == null) return ErrorCode.EntityNull;
            if (bo == null) return ErrorCode.BoNull;
            entity.Name = bo.Name;
            entity.SortOrder = bo.SortOrder;
            entity.IsDefault = bo.IsDefault;
            if (bo.UnitId != 0) entity.UnitId = bo.UnitId;
            return ErrorCode.Success;
        }
    }
}

File: Service/Translators/UnitFinishingOptionTranslator.vb
Imports BO
Imports DAL

Public Class UnitFinishingOptionTranslator
    Public Shared Function TranslateEntityToBO(_entity As UnitFinishingOption, bo As UnitFinishingOptionBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        bo.Id = _entity.Id
        bo.UnitId = _entity.UnitId
        bo.Name = _entity.Name
        bo.SortOrder = _entity.SortOrder
        bo.IsDefault = _entity.IsDefault
        For Each cv In _entity.UnitConstructionValues
            Dim bocv As New UnitConstructionValueBO
            Dim err = UnitConstructionValueTranslator.TranslateEntityToBO(cv, bocv)
            If err = ErrorCode.Success Then
                bo.ConstructionValues.Add(bocv)
            End If
        Next
        Return ErrorCode.Success
    End Function
End Class
